homeassistant/helpers/storage.py

```diff
@@ -94,6 +93,7 @@ def path(self):
         """
         if self._load_task is None:
             self._load_task = self.hass.async_add_job(self._async_load())
```

Comment:
If we would use `self.hass.async_create_task` here instead of `self.hass.async_add_job` we wouldn't need the assert below.